internal/utils/utils.go

```diff
@@ -529,3 +529,38 @@ func GetConfigValue(data map[string]string, key, def string) string {
 	}
 	return def
 }
+
+// ParseDurationFromConfig parses a duration string from ConfigMap with default fallback
+// Returns the parsed duration or the default value if parsing fails or key is missing
+func ParseDurationFromConfig(data map[string]string, key string, defaultValue time.Duration) time.Duration {
```

Comment:
Go encourages (best-practice) this function (and in general all config related functions) to be stored in a file called `config.go`

Reply:
done